### PR TITLE
tweak clang-format configuration

### PR DESCRIPTION
**Pull request by tamiko**
In this PR I will collect all clang-format configuration tweaks we have
discussed and that got positive feedback.

### CODE REVIEW

**Comment by tamiko** [comment]
@dealii/dealii Shall we merge this? I have the feeling that the discussion pretty much died down and everyone agreed to these changes. Further, it really removes a couple of ugly indenting corner cases :-)

File: .clang-format [hunk 1]
```diff
@@ -45,14 +45,13 @@ BreakBeforeBinaryOperators: None
 BreakBeforeBraces: Custom
 BreakBeforeInheritanceComma: false
 BreakBeforeTernaryOperators: false
-BreakConstructorInitializers: AfterColon
+BreakConstructorInitializersBeforeComma: true
```
Comment (masterleinad) [inline .clang-format]: Would you mind changing this to `BreakConstructorInitializers: BeforeComma`?

File: examples/.clang-format [hunk 2]
```diff
@@ -47,14 +47,13 @@ BreakBeforeBinaryOperators: None
 BreakBeforeBraces: Custom
 BreakBeforeInheritanceComma: false
 BreakBeforeTernaryOperators: false
-BreakConstructorInitializers: AfterColon
+BreakConstructorInitializersBeforeComma: true
```
Comment (masterleinad) [inline examples/.clang-format]: Would you mind changing this to `BreakConstructorInitializers: BeforeComma`?

**Comment by bangerth** [comment]
Yes, I like this better.

**Comment by tamiko** [comment]
@masterleinad fixed.

**Pull request merged**